engine/storage/src/main/java/org/apache/cloudstack/storage/endpoint/DefaultEndPointSelector.java

```diff
@@ -318,6 +319,35 @@ protected EndPoint findEndpointForImageStorage(DataStore store) {
         return RemoteHostEndPoint.getHypervisorHostEndPoint(host);
     }
 
+    protected Host getVmwareHostFromVolumeToDelete(VolumeInfo volume) {
+        VirtualMachine vm = volume.getAttachedVM();
+        if (vm == null) {
```

Comment:
If the volume isn't attached to a VM - would it be removed by GC (if not by this code) then?

Reply:
Yes then the calling method code tries to find a host based on the storage at line 476 https://github.com/apache/cloudstack/pull/7312/files#diff-45668a1459e79e7705bb54a8f6e09d972a12ffe972b06b61468e06f4f8ce2c3dR476